Add recipe for cinspect-mode

inlinestyle opened the pull request:
Hi! I'd like to add a recipe for [cinspect-mode](https://github.com/inlinestyle/cinspect-mode).
 - `cinspect-mode` gives an emacs user easy access to the [cinspect](https://github.com/punchagan/cinspect) project (source code inspection for CPython builtins).
 - I am the creator and maintainer. 
 - I've tested and verified that the package builds and installs in the MELPA sandbox

Please let me know if there's any other info you need.

purcell commented:
Thanks. The `:files` spec is redundant here because the default value excludes `*-test.el` by default. So please can you amend the recipe commit to remove `:files` and force-push it back here?

There are a few issues with the code itself too:
- Don't autoload the change to `python-mode-hook`: the user should decide whether to enable installed packages, just as with all the packages pre-installed with Emacs
- Since this is a minor-mode, the `-mode` prefix for the package/feature name is frowned upon. Ideally this file/feature/package should simply be called `cinspect`/`cinspect.el`.
- Use the `cinspect-` prefix for public identifiers and `cinspect--` for internal symbols in order to conform to the [elisp coding conventions](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html). Whatever you may have seen elsewhere (`:`, `/` etc.), following the conventions is the best way to help the elisp community.

Hope that helps!

-Steve

inlinestyle commented:
Hi Steve,
Thanks for the advice! I've implemented the three changes you mentioned to the `cinspect` package. I've also adjusted this PR to respect the name change, as well as the `:file` convention you mentioned.

purcell commented:
Super - merged, thanks1
